js/ui/grid_core/ui.grid_core.editing.js

```diff
@@ -263,7 +263,7 @@ var EditingController = modules.ViewController.inherit((function() {
                 columnIndex;
 
             if(getEditMode(this) === EDIT_MODE_FORM && this._firstFormItem) {
-                columnIndex = this._firstFormItem.column.index;
+                columnIndex = this._firstFormItem.column.visibleIndex;
```

Comment:
visibleIndex is not a real visible index. use columnsController.getVisibleIndex method to get real visible index